src/theory/arith/arith_utilities.h

```diff
@@ -302,6 +301,31 @@ inline Node mkPi()
   return NodeManager::currentNM()->mkNullaryOperator(
       NodeManager::currentNM()->realType(), kind::PI);
 }
+/** Join kinds, where k1 and k2 are arithmetic relations returns an
+ * arithmetic relation ret such that
+ * if (a <k1> b) and (a <k2> b), then (a <ret> b).
+ */
+Kind joinKinds(Kind k1, Kind k2);
```

Comment:
What it is doing -- What is the contract of this function?

Reply:
If I understood correctly from the code, Is it getting rid of redundant relation?

Reply:
This is used in the lemma schema here: https://github.com/CVC4/CVC4/blob/master/src/theory/arith/nonlinear_extension.h#L828.

Basically, if you have two relations between related terms, you need to know what the resulting relation is.